### ISSUE TITLE
Low coverage

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#2590]
We are currently one branch away from hitting coverage threshold. I think that's a bit too low, and might prevent people from creating pull requests if they can't pass `npm test`.

**Comment by lo1tuma** [comment]
If this issue is only about allowing people to contribute code with bad coverage then I vote for 'won't fix'.

I agree that the coverage could be increased but then we should also increase our thresholds for `check-coverage`.

**Comment by ilyavolodin** [comment]
@lo1tuma You are not taking into account branch coverage vs. statement and line coverage. I don't want to create a buffer for line/statement coverage, but brunch coverage is different. It's actually very easy to create fully tested code that still misses a branch or two here and there. We also had a pretty significant discrepancy between line coverage and branch coverage (branch coverage was quite a bit lower then line and statement coverages).

**Comment by lo1tuma** [comment]
How is branch coverage different? And why do you call code "fully tested" if a coverage tool shows you that you missed one ore more branches?

**Comment by michaelficarra** [comment]
@ilyavolodin not only is that insufficient, but even if you had full branch coverage, you should also consider all paths through branches and all possible values at each branch. Partial branch coverage is *incredibly* far from "fully tested" code, at least in my book.

**Comment by ilyavolodin** [comment]
Just an overly simplified example:
```js
var a = function(b) {
  b = b || {};
  console.log(b);
}
```
If you just cover happy path (pass in parameter), one branch out of two is going to be uncovered. It isn't fully tested code (I agree), however, do you really have to test unhappy path, and does that test provide any value? I know it's over-simplified code, but still. In a lot of cases, branches are created as defense mechanism, that will not ever get hit. It's just a sanity check. I did add a few tests that cover those sanity checks, but I'm not sure they provide any value (like the one I added for ignore-path file).

**Comment by nzakas** [comment]
Yes, you have to read unhappy path - that's usually where bugs hide.

**Comment by nzakas** [comment]
Er, test unhappy path
